### ISSUE TITLE
[expo-build-properties] Plugin does not update `iOS Deployment Target` of Project

### ISSUE THREAD

**Issue by tjcarrel** [expo/expo#30610]
### Minimal reproducible example

https://github.com/tjcarrel/expo-build-properties-example

### What platform(s) does this occur on?

iOS

### Where did you reproduce the issue?

in a development build

### Summary

Using the `expo-build-properties` plugin, the `iOS Deployment Target` for the app's target gets changed, but not the project overall. Desired behavior is the Deployment Target gets changed for the project as well.

1. Run `npx expo prebuild -p ios --clean`
2. Check the deployment version of the project.

Here is my `app.json`:

```json
{
  "expo": {
    "name": "my-app",
    "slug": "my-app",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/images/icon.png",
    "scheme": "myapp",
    "userInterfaceStyle": "automatic",
    "splash": {
      "image": "./assets/images/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "com.tjcarrel.my-app"
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/images/adaptive-icon.png",
        "backgroundColor": "#ffffff"
      }
    },
    "web": {
      "bundler": "metro",
      "output": "static",
      "favicon": "./assets/images/favicon.png"
    },
    "plugins": [
      "expo-router",
      [
        "expo-build-properties",
        {
          "ios": {
            "deploymentTarget": "16.0"
          }
        }
      ]
    ],
    "experiments": {
      "typedRoutes": true
    }
  }
}
```

### Environment
```bash
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.15.0 - ~/.nvm/versions/node/v20.15.0/bin/node
      npm: 10.7.0 - ~/.nvm/versions/node/v20.15.0/bin/npm
      Watchman: 2024.07.01.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.10121639
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.22 => 51.0.22
      expo-router: ~3.5.18 => 3.5.18
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.74.3 => 0.74.3
      react-native-web: ~0.19.10 => 0.19.12
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```bash
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check for common project setup issues
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check native tooling versions
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!
```

**Comment by Kudo** [comment]
hi there! what targets than the app target that you expect to change for the ios version? e.g. do you have app extensions or some else targets?

**Comment by cemck** [comment]
i have the same issue, I get an error for my native module, although I specified iOS 14.0 in config file with expo-build-properties: (targets in Xcode workspace are set properly)

```
      [
        "expo-build-properties",
        {
          ios: {
            deploymentTarget: "14.0",
          }
        }
      ]
```

```
  85 |     Function("endActivity") { (title: String, calcutatedResult: String) -> Void in
  86 |         let logger = Logger()
> 87 |         logger.info("endActivity()")
     |                     ^ 'OSLogMessage' is only available in iOS 14.0 or newer
  88 |         
```

**Comment by cemck** [comment]
I found the problem was because the plugin module has a seperate iOS version specified in `modules/MODULENAME/ios/MODULENAME.podspec`
updating the deployment in this file fixed it for me
